### PR TITLE
Task #231201 generate order id

### PR DESCRIPTION
**Pull request by namita-25**
add iframe 
and generate order id

## Summary by CodeRabbit

- **New Features**
	- Enhanced iframe communication for improved data handling during form submissions.
	- Added loading indicators to enhance user experience while data is processed.
	- Introduced a new data transformation function for better handling of user data.

- **Bug Fixes**
	- Improved error handling during form submission confirmation.

- **Refactor**
	- Restructured data types and state management for better clarity and efficiency in the `BenefitsDetails` component. 
	- Simplified the logic for managing user and context data.

- **Documentation**
	- Updated interfaces for better alignment with application data structures.

### CODE REVIEW

File: src/components/WebView.tsx [hunk 1]
```diff
@@ -1,118 +1,205 @@
...
   const navigate = useNavigate();
 
-  const searchForm = async (url: string) => {
-    try {
-      const response = await fetch(url, { method: "GET" });
-      if (!response.ok) {
-        throw new Error(`HTTP error! Status: ${response.status}`);
+  useEffect(() => {
+    const handleMessage = (event: MessageEvent) => {
+      if (event.origin !== "http://localhost:5174") {
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/components/WebView.tsx:98]: @namita-25 does this need config?

File: src/components/WebView.tsx [hunk 2]
```diff
@@ -1,118 +1,205 @@
...
+        return;
       }
+      if (event.data.type === "FORM_SUBMIT") {
+      }
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/components/WebView.tsx:102]: @namita-25 if block is empty

File: src/screens/benefit/Details.tsx [hunk 3]
```diff
@@ -114,45 +123,46 @@ const BenefitsDetails: React.FC = () => {
...
+  //       setConfirmationConsent({ orderId, name: item?.descriptor?.name });
+  //     } else {
+  //       setError("Error while creating application. Please try again later");
+  //     }
+  //   } catch (e: unknown) {
+  //     setError(`Error: ${(e as Error).message}`);
+  //   } finally {
+  //     setLoading(false);
+  //   }
+  // };
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/screens/benefit/Details.tsx:165]: use /**/ for multiline comment

File: src/screens/benefit/Details.tsx [hunk 4]
```diff
@@ -49,35 +48,45 @@ interface BenefitItem {
...
+  ttl: string;
+  timestamp: string;
+}
 interface AuthUser {
   user_id?: string;
   name?: string;
-  class?: string;
-  previousYearMarks?: string;
-  phoneNumber?: string;
+  current_class?: string;
```
Comment (manojLondhe) [inline src/screens/benefit/Details.tsx:69]: class?: string;
  previousYearMarks?: string;
  phoneNumber?: string;
  
  or

  current_class?: string;
previous_year_marks?: string;
  phone_number?: string;

File: src/components/WebView.tsx [hunk 5]
```diff
@@ -1,118 +1,205 @@
...
-      searchForm(url);
+  const submitConfirm = async (payload) => {
+    const confirmPayload = {
+      submission_id: payload?.submit.submission_id,
+      item_id: payload?.submit.application_id,
+      benefit_id: id,
+      context: context,
+    };
+    const result = await confirmApplication(confirmPayload);
+    setIsLoading(true);
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/components/WebView.tsx:146]: @namita-25 fix this

**Comment by namita-25** [comment]
used in solve-coderabit branch

**Pull request closed without merging**